internal/config/config.go

```diff
@@ -142,7 +142,9 @@ type BaseStep struct {
 }
 
 type CloneStep struct {
-	BaseStep `json:",inline"`
+	BaseStep          `json:",inline"`
+	Depth             int  `json:"depth"`
+	RecurseSubmodules bool `json:"recursesubmodules"`
```

Comment:
Should be `recurse_submodules`

Reply:
Should be good to add depth validation to check that the provided depth value is > 0

Reply:
> 
> 
> Should be `recurse_submodules`

Done.
> 
> 
> Should be good to add depth validation to check that the provided depth value is > 0

I had already added this check in the function:
https://github.com/camandel/agola/blob/191cecee120e276ae554058b25b9d35eb0a86c41/internal/runconfig/runconfig.go#L449-L451

but if depth value is not valid the parameter is simply ignored silently.
To raise an error during the "Task setup" step, if json can't be parsed, I could add a field validator like this: 
```
        Depth             int  `json:"depth" validate:"min=1"`
```
but for that it needs to add a new external library (validator.v2).
Not sure if it's what you asked or if you thought about a simpler and clearer way to achieve that. Thanks

Reply:
No need for another library, just make it a `*int`. Then you can check if it's not provided (nil) or provided and verify it's value in the verify config func. The same should be used to choose if to define the --depth option or not.

Reply:
Used *int as type for depth and added check in the checkConfig function.